nessie/jobs/generate_merged_student_feeds.py

```diff
@@ -97,14 +97,14 @@ def generate_feeds(self):
         feed_path = app.config['LOCH_S3_BOAC_ANALYTICS_DATA_PATH'] + '/feeds/'
         s3.upload_json(advisees_by_canvas_id, feed_path + 'advisees_by_canvas_id.json')
 
-        # Avoid processing Canvas analytics data for future terms and pre-CS terms.
-        for term_id in future_term_ids():
+        term_ids = reverse_term_ids(include_future_terms=True, include_legacy_terms=True)
+        for term_id in term_ids:
             GenerateMergedEnrollmentTerm().refresh_student_enrollment_term(term_id, enrollment_terms_map.get(term_id, {}))
+            s3.upload_json(enrollment_terms_map.get(term_id, {}), feed_path + f'enrollment_term_map_{term_id}.json')
```

Comment:
@lyttam This is tricky, but there might be some redundant work going on here. I think we need to explicitly call `GenerateMergedEnrollmentTerm().refresh_student_enrollment_term` for future terms and legacy terms, but not for those terms included in our usual `reverse_term_ids()`, because those terms will get refreshed with analytics as a consequence of the `queue_merged_enrollment_term_jobs` call below.

Conversely, the `s3.upload_json(enrollment_terms_map...` only needs to happen for our usual `reverse_term_ids()`. We stash that data in S3 so that worker nodes can pick it up, and it's only our expensive analytics-containing terms that are distributed among worker nodes by the `queue_merged_enrollment_term_jobs` logic.

Reply:
Thanks @paulkerschen, I wasn't quite sure about that bit. Will amend.